saleor/data_feeds/google_merchant.py

```diff
@@ -0,0 +1,219 @@
+from __future__ import unicode_literals
+
+import gzip
+import csv
+from os import path
+
+from django.conf import settings
+from django.contrib.sites.models import Site
+from django.contrib.syndication.views import add_domain
+from django.core.files.storage import default_storage
+
+from ..discount.models import Sale
+from ..product.models import ProductVariant, Category
+
+CATEGORY_SEPARATOR = ' > '
+
+FILE_PATH = path.join(settings.INTEGRATIONS_DIR, 'saleor-feed.csv.gz')
+FILE_URL = default_storage.url(FILE_PATH)
+COMPRESSION = False
+
+ATTRIBUTES = ['id', 'title', 'product_type', 'google_product_category',
+              'link', 'image_link', 'condition', 'availability',
+              'price', 'tax', 'shipping', 'sale_price',
+              'mpn', 'brand', 'item_group_id', 'gender', 'age_group',
+              'color', 'size', 'description']
+
+
+def get_feed_items():
+    items = ProductVariant.objects.all()
+    items = items.select_related('product')
+    items = items.prefetch_related(
+        'images', 'stock', 'product__attributes', 'product__categories',
+        'product__images',
+    )
+    return items
+
+
+def item_id(item):
+    return item.sku
+
+
+def item_mpn(item):
+    return str(item.sku)
+
+
+def item_guid(item):
+    return item.sku
+
+
+def item_link(item, current_site):
+    return add_domain(current_site.domain,
+                      item.get_absolute_url(),
+                      not settings.DEBUG)
+
+
+def item_title(item):
+    return item.display_product()
+
+
+def item_description(item):
+    return item.product.description[:100]
+
+
+def item_condition(item):
+    """
+    Allowed values: new, refurbished, or used
+    Read more:
+    https://support.google.com/merchants/answer/6324469
+    """
+    return 'new'
+
+
+def item_brand(item):
+    """
+    This field is required.
+    Read more:
+    https://support.google.com/merchants/answer/6324351?hl=en&ref_topic=6324338
+    """
+    return 'Saleor'
+
+
+def item_tax(item, discounts):
+    """
+    For some countries you need to set tax info
+    Read more:
+    https://support.google.com/merchants/answer/6324454
+    """
+    price = item.get_price_per_item(discounts=discounts)
+    return 'US::%s:y' % price.tax
+
+
+def item_group_id(item):
+    return str(item.product.pk)
+
+
+def item_image_link(item, current_site):
+    image = item.get_first_image()
+    if image:
+        return add_domain(current_site.domain, image.url, False)
+    else:
+        return None
+
+
+def item_availability(item):
+    if item.get_stock_quantity():
+        return 'in stock'
+    else:
+        return 'out of stock'
+
+
+def item_shipping(item):
+    """
+    You can set one shipping cost in feed settings or precise shipping cost
+    for every item.
+    Read more:
+    https://support.google.com/merchants/answer/7050921
+    """
+    return ''
```

Comment:
Doesn't this render this feed useless? We have per-country shipping methods available in the database.